### PR TITLE
#105 Added storage_backend configuration element

### LINKED ISSUE #105: Add configuration element - backend
The user should choose the type of backend - onprem or saas. The notebook connector should provide a new configuration element for that.

This makes the backend inference obsolete.

### LINKED ISSUE #106: Update the logic of the connection.get_udf_bucket_path function
It should now use the bucket's `udf_path` property.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #105 
closes #106

### CODE REVIEW

File: exasol/nb_connector/connections.py [hunk 1]
```diff
@@ -154,7 +83,8 @@ def get_udf_bucket_path(conf: Secrets) -> str:
     Builds the path of the BucketFS bucket specified in the configuration,
     as it's seen in the udf's file system.
     """
-    return f"/buckets/{conf.get(CKey.bfs_service)}/{conf.get(CKey.bfs_bucket)}"
+    bucket = open_bucketfs_connection(conf)
+    return bucket.udf_path
```
Comment (tomuben) [inline exasol/nb_connector/connections.py:87]: Wouldn't the return type "Pathlike" would fit better for this function, instead of 'str'?
Sorry if this question is not related to the changes in the PR.

**Review by tomuben — APPROVED** [review]

File: exasol/nb_connector/connections.py (hunk 1 above)
Comment (ahsimb) [reply exasol/nb_connector/connections.py:87]: Maybe, but then I would break compatibility and will have to change few notebooks in the AI-Lab.
